File: TA5/tool-evaluation-4/HAMR/examples/test_data_port_periodic_domains_VM/sender_vm/CAmkES_seL4/src/c/ext-c/consumer_t_impl_dst_process_dst_thread/consumer_t_impl_dst_process_dst_thread.c
#include <consumer_t_impl_dst_process_dst_thread_api.h>
#include <consumer_t_impl_dst_process_dst_thread.h>
#include <ext.h>

// This file will not be overwritten so is safe to edit

Unit sender_vm_test_event_data_port_periodic_domains_consumer_t_impl_dst_process_dst_thread_initialise_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "consumer_t_impl_dst_process_dst_thread.c", "", "sender_vm_test_event_data_port_periodic_domains_consumer_t_impl_dst_process_dst_thread_initialise_", 0);

  // examples of api setter and logging usage

  api_logInfo__sender_vm_test_event_data_port_periodic_domains_consumer_t_impl_dst_process_dst_thread(SF string("Example logInfo"));

  api_logDebug__sender_vm_test_event_data_port_periodic_domains_consumer_t_impl_dst_process_dst_thread(SF string("Example logDebug"));

  api_logError__sender_vm_test_event_data_port_periodic_domains_consumer_t_impl_dst_process_dst_thread(SF string("Example logError"));
}

Unit sender_vm_test_event_data_port_periodic_domains_consumer_t_impl_dst_process_dst_thread_finalise_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "consumer_t_impl_dst_process_dst_thread.c", "", "sender_vm_test_event_data_port_periodic_domains_consumer_t_impl_dst_process_dst_thread_finalise_", 0);
}

Unit sender_vm_test_event_data_port_periodic_domains_consumer_t_impl_dst_process_dst_thread_timeTriggered_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "consumer_t_impl_dst_process_dst_thread.c", "", "sender_vm_test_event_data_port_periodic_domains_consumer_t_impl_dst_process_dst_thread_timeTriggered_", 0);

  // examples of api getter usage

  S8 t0;
  if(api_get_read_port__sender_vm_test_event_data_port_periodic_domains_consumer_t_impl_dst_process_dst_thread(SF &t0)) {
    DeclNewString(read_port_str);
    String__append(SF (String) &read_port_str, string("Received on read_port: "));
    S8_string_(SF (String) &read_port_str, t0);
    api_logInfo__sender_vm_test_event_data_port_periodic_domains_consumer_t_impl_dst_process_dst_thread(SF (String) &read_port_str);
  }
}

File: TA5/tool-evaluation-4/HAMR/examples/test_data_port_periodic_domains_VM/both_vm/CAmkES_seL4/src/main/component/both_vm/test_event_data_port_periodic_domains/consumer_t_impl_dst_process_dst_thread.scala
// #Sireum

package both_vm.test_event_data_port_periodic_domains

import org.sireum._
import both_vm._

// This file will not be overwritten so is safe to edit
object consumer_t_impl_dst_process_dst_thread {

  def initialise(api: consumer_t_impl_Initialization_Api): Unit = {
    // example api usage

    api.logInfo("Example info logging")
    api.logDebug("Example debug logging")
    api.logError("Example error logging")

  }

  def timeTriggered(api: consumer_t_impl_Operational_Api): Unit = {
    // example api usage

    val apiUsage_read_port: Option[Base_Types.Integer_8] = api.get_read_port()
    api.logInfo(s"Received on read_port: ${apiUsage_read_port}")
  }

  def activate(api: consumer_t_impl_Operational_Api): Unit = { }

  def deactivate(api: consumer_t_impl_Operational_Api): Unit = { }

  def finalise(api: consumer_t_impl_Operational_Api): Unit = { }

  def recover(api: consumer_t_impl_Operational_Api): Unit = { }
}
